contracts/axelar-gateway/src/contract.rs

```diff
@@ -232,4 +253,178 @@ impl AxelarGateway {
             .get(&DataKey::MessageApproval(key))
             .unwrap_or(MessageApprovalValue::NotApproved)
     }
+
+    fn message_approval_hash(env: &Env, message: Message) -> MessageApprovalValue {
+        MessageApprovalValue::Approved(env.crypto().keccak256(&message.to_xdr(env)).into())
+    }
+
+    fn validate_proof(env: &Env, data_hash: BytesN<32>, proof: Proof) -> bool {
```

Comment:
Move all the auth functionality to a separate `auth.rs` module in this crate

Reply:
done